src/parameters/ByteArrayParameter.cpp

```diff
@@ -37,7 +37,7 @@ namespace sql
   void ByteArrayParameter::writeTo(SQLString& str)
   {
     str.append(BINARY_INTRODUCER);
-    Utils::escapeData(bytes.arr, static_cast<size_t>(bytes.length>0?bytes.length:-bytes.length), noBackslashEscapes, str);
+    Utils::escapeData(bytes.arr, static_cast<size_t>(bytes.size()), noBackslashEscapes, str);
```

Comment:
bytes.size() returns size_t, so the cast is not really needed. but not a bit problem